akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala

```diff
@@ -1171,38 +1167,31 @@ class DDataShardCoordinator(
       else
         context.become(waitingForUpdate(evt, afterUpdateCallback, newRemainingKeys))
 
-    case UpdateTimeout(AllShardsKey, Some(newShard: String)) =>
-      if (terminated) {
-        log.error(
-          "The ShardCoordinator was unable to update shards distributed state within 'updating-state-timeout': {} millis (terminating), event={}",
-          writeMajority.timeout.toMillis,
-          evt)
-        onTerminated()
+    case UpdateTimeout(AllShardsKey, Some(_: String)) =>
+      log.error(
+        "The ShardCoordinator was unable to update shards distributed state within 'updating-state-timeout': {} millis ({}), event={}",
+        writeMajority.timeout.toMillis,
+        if (terminating) "terminating" else "retrying",
+        evt)
+      if (terminating) {
+        context.stop(self)
       } else {
-        log.error(
-          "The ShardCoordinator was unable to update shards distributed state within 'updating-state-timeout': {} millis (retrying), event={}",
-          writeMajority.timeout.toMillis,
-          evt)
         // repeat until UpdateSuccess
-        sendAllShardsUpdate(newShard)
+        sendCoordinatorStateUpdate(evt)
```

Comment:
is this change correct?
it's `UpdateTimeout(AllShardsKey`

Reply:
It is not correct. Fixed